### PR TITLE
Named tracks in oncoprinter. This allows duplicate gene tracks and combined gene tracks

### PR DESCRIPTION
**Pull request by adamabeshouse**
The main use case will be using the "Open in Oncoprinter" feature from the results view.

[Combined gene track](https://www.cbioportal.org/results/oncoprint?Action=Submit&RPPA_SCORE_THRESHOLD=2.0&Z_SCORE_THRESHOLD=2.0&cancer_study_id=coadread_tcga_pub&cancer_study_list=coadread_tcga_pub&case_set_id=coadread_tcga_pub_nonhypermut&data_priority=0&gene_list=%255BKRAS%2520NRAS%2520BRAF%255D&gene_set_choice=user-defined-list&geneset_list=%20&genetic_profile_ids_PROFILE_COPY_NUMBER_ALTERATION=coadread_tcga_pub_gistic&genetic_profile_ids_PROFILE_MUTATION_EXTENDED=coadread_tcga_pub_mutations&profileFilter=0&tab_index=tab_visualize)
image

[Duplicate gene tracks](https://www.cbioportal.org/results/oncoprint?Action=Submit&RPPA_SCORE_THRESHOLD=2.0&Z_SCORE_THRESHOLD=2.0&cancer_study_id=coadread_tcga_pub&cancer_study_list=coadread_tcga_pub&case_set_id=coadread_tcga_pub_nonhypermut&data_priority=0&gene_list=PTEN%253AMUT%2520%250APTEN%253AAMP%2520HOMDEL&gene_set_choice=user-defined-list&geneset_list=%20&genetic_profile_ids_PROFILE_COPY_NUMBER_ALTERATION=coadread_tcga_pub_gistic&genetic_profile_ids_PROFILE_MUTATION_EXTENDED=coadread_tcga_pub_mutations&profileFilter=0&tab_index=tab_visualize)
image

### CODE REVIEW

File: src/pages/staticPages/tools/oncoprinter/OncoprinterGeneticUtils.ts [hunk 1]
```diff
@@ -93,6 +93,7 @@ export type OncoprinterGeneticInputLineType2 = OncoprinterGeneticInputLineType1
         | 'mrnaLow'
         | 'protHigh'
         | 'protLow';
+    trackLabel?: string;
```
Comment (onursumer) [inline src/pages/staticPages/tools/oncoprinter/OncoprinterGeneticUtils.ts:96]: Should we name this property `trackName` as the PR title suggests? :)

File: src/pages/staticPages/tools/oncoprinter/OncoprinterImportUtils.ts [hunk 2]
```diff
@@ -140,25 +142,47 @@ function getOncoprinterGeneticInputLine(parsed: OncoprinterGeneticInputLine) {
...
     sampleOrPatient: 'sample' | 'patient'
 ) {
-    const parsedLines = [];
-    for (const oncoprintDatum of oncoprintData) {
-        parsedLines.push(
-            ...oncoprintDatum.data.map(d => {
-                return getOncoprinterParsedGeneticInputLine(d, oncoprintDatum[
-                    sampleOrPatient
-                ] as string);
+    // handle case where tracks have same label
```
Comment (onursumer) [inline src/pages/staticPages/tools/oncoprinter/OncoprinterImportUtils.ts:160]: This looks like a good candidate to isolate into a utility function (`generateUniqNames`?) and add a few tests.

Comment (dippindots) [reply src/pages/staticPages/tools/oncoprinter/OncoprinterImportUtils.ts:160]: Agree!

**Comment by adamabeshouse** [comment]
@dippindots @onursumer thanks for the review! please check it out again when you can

**Comment by tmazor** [comment]
Looks good @adamabeshouse ! The only thing I noticed is that gene tracks made up of multiple genes don't have an "Expand" option. This is a useful feature in OncoPrint so if feasible, it would be nice to have this in OncoPrinter also.

**Comment by adamabeshouse** [comment]
@tmazor good idea but lets put that off until later, that's a whole other project https://github.com/cBioPortal/cbioportal/issues/7717

**Pull request merged**